### ISSUE TITLE
[PLACEHOLDER] Tournament-V5 Bugs/Issues

### ISSUE THREAD

**Issue by tisbells** [RiotGames/developer-relations#828]
**Placeholder for any issues with Tournament-V5**  
For Tournament-V5 once live - Please reply to this issue with any technical problems or bugs with the new API.

**Comment by Pikeman** [comment]
We're getting HTTP 500 errors from `POST https://americas.api.riotgames.com/lol/tournament/v5/providers` now. Our requests were successful in testing yesterday though.

**Comment by tisbells** [comment]
@Pikeman Should be fixed, could you please verify?

**Comment by Pikeman** [comment]
> @Pikeman Should be fixed, could you please verify?

It works! Thanks for the quick fix.

**Comment by Pikeman** [comment]
We're getting null PUUID on LobbyEventV5DTO:
```json
[
  {
    "timestamp": "2023-10-12 01:14:35.0",
    "eventType": "PracticeGameCreatedEvent",
    "puuid": null
  },
  {
    "timestamp": "2023-10-12 01:14:56.0",
    "eventType": "PlayerJoinedGameEvent",
    "puuid": null
  },
  {
    "timestamp": "2023-10-12 01:15:04.0",
    "eventType": "PlayerJoinedGameEvent",
    "puuid": null
  },
  ...
```

**Comment by MakeevD** [comment]
Hey there!

We are not receiving callbacks after the finished match. The same issue was with V4 API before, and we use an HTTP endpoint with a correct gTLD.

Could you please tell what can be an issue?

**Comment by tisbells** [comment]
Thanks for your patience, we're looking into the above issues and will get back to you soon.

**Comment by tisbells** [comment]
@Pikeman This should be resolved, please let us know if not.
@MakeevD If you could send me your app ID thru Discord or message us in the app notes we'd appreciate it so we can look at the games you're not getting callbacks for. We're looking at callback issues in general, but seeing specific examples is very helpful.

**Comment by ElijahKSmith** [comment]
> Hey there!
> 
> We are not receiving callbacks after the finished match. The same issue was with V4 API before, and we use an HTTP endpoint with a correct gTLD.
> 
> Could you please tell what can be an issue?

+1 on this. @tisbells I can also share the same info if it helps!

**Comment by tisbells** [comment]
For those experiencing callback issues - "Region" was unexpectedly swapped to "shardId" which may be causing errors for multi-region tournaments. This should be changed back to "Region" on our end early next week. Additionally, you will not receive a callback if the match did not finish to completion or to surrender.

**Comment by seju-play-vs** [comment]
For anyone else having problems, if there are any bots in the game, you won't get a callback.

**Comment by tisbells** [comment]
`shardId `was swapped back to `region`. Additionally, an issue was found that may have caused some errors for EUW tournaments and this has been fixed.

**Comment by DaBenjamins** [comment]
When the callback is fixed would I need to redo my URL? its been down for months.

**Comment by ElijahKSmith** [comment]
> When the callback is fixed would I need to redo my URL? its been down for months.

You've gotta create a new Provider for V5 if you haven't already

**Comment by DaBenjamins** [comment]
> > When the callback is fixed would I need to redo my URL? its been down for months.
> 
> You've gotta create a new Provider for V5 if you haven't already

Was that documented somewhere as a requirement, my old Provider and Tournament IDs kept working on V5

**Comment by seju-play-vs** [comment]
> > When the callback is fixed would I need to redo my URL? its been down for months.
> 
> You've gotta create a new Provider for V5 if you haven't already

@tisbells Is this true? I did not create a new provider, and our v5 code is working.

**Comment by ElijahKSmith** [comment]
> > > When the callback is fixed would I need to redo my URL? its been down for months.
> > 
> > 
> > You've gotta create a new Provider for V5 if you haven't already
> 
> Was that documented somewhere as a requirement, my old Provider and Tournament IDs kept working on V5
> 
> Update: If I try and get a new Provider ID, it just gives me the old one since its the same URL, so I dont see how to get a new ID

Sorry I should've included more information in my original reply to explain why I said that. I meant to say that if it still wasn't working you should try to generate a new one. For my own providers I found that one provider when re-registered with the same information returned the same ID and worked the same, but on two other ones I got new IDs which were needed to make it work. Not sure why that's the case but I figured it might be worth checking.

**Comment by tisbells** [comment]
You should be able to use your original provider with V5.

**Comment by DaBenjamins** [comment]
> You should be able to use your original provider with V5.

any new on when Callback will be working again?

**Comment by tisbells** [comment]
> > You should be able to use your original provider with V5.
> 
> any new on when Callback will be working again?

@DaBenjamins If you are still experiencing callback issues after finishing a match to completion/surrender, please send us an app message in the developer portal with your callback URL and a recent tournament code.

**Comment by tisbells** [comment]
Reposting this here from https://github.com/RiotGames/developer-relations/issues/833 - 
We now have a new endpoint for getting the PUUIDs for Tournaments! https://developer.riotgames.com/apis#tournament-v5/GET_getGames

This will provide the same payload of tournament callbacks + participant PUUIDs. You can also use this endpoint to check if the game was recorded and help debug callbacks. If the endpoint returns the game, it means a callback was attempted. Please let us know if any of you run into problems with this.

**Comment by OllieJennings** [comment]
@tisbells i noticed one thing, not sure if intentional or a bug, but when you request the tournament game by code, the `metaData` seems to be empty.

The server callback had correct `metaData`, and also requesting the tournamentDTO returns the correct `metaData`.

**Comment by R0kiee** [comment]
I dont know if it is how it's work, but im getting trouble to make a tournament (i dont need a web page to see my results (can i use localhost), im testing only to create little tournaments by myself)
![Captura de pantalla 2023-11-20 034010](https://github.com/RiotGames/developer-relations/assets/74803185/8e143c41-435b-4a99-a223-bb614357dd29)

**Comment by Token07** [comment]
@R0kiee the tournament API is only deployed to americas. Any tournament created on americas is valid for all regions; This has always been how the tournament API works.

**Comment by R0kiee** [comment]
> @R0kiee the tournament API is only deployed to americas. Any tournament created on americas is valid for the region set in the provider; This has always been how the tournament API works.

And something about the other question??

**Comment by Token07** [comment]
@R0kiee I don't believe the URL needs to be valid. It looks like your attempt to create a provider was successful so now you create a tournament using the providerId returned.

**Comment by R0kiee** [comment]
After make the first stop i got a HTTP 500 from [POST]/lol/tournament-stub/v5/codes]

**Comment by tisbells** [comment]
> @tisbells i noticed one thing, not sure if intentional or a bug, but when you request the tournament game by code, the `metaData` seems to be empty.
> 
> The server callback had correct `metaData`, and also requesting the tournamentDTO returns the correct `metaData`.

@OllieJennings Can you please provide a tournament code where you're seeing this?

**Comment by R0kiee** [comment]
> > @tisbells i noticed one thing, not sure if intentional or a bug, but when you request the tournament game by code, the `metaData` seems to be empty.
> > The server callback had correct `metaData`, and also requesting the tournamentDTO returns the correct `metaData`.
> 
> @OllieJennings Can you please provide a tournament code where you're seeing this?

I dont have server callback (i use test on this field), i only want to test getting a tournament code, i dont know if it makes some issues

**Comment by OllieJennings** [comment]
> > @tisbells i noticed one thing, not sure if intentional or a bug, but when you request the tournament game by code, the `metaData` seems to be empty.
> > The server callback had correct `metaData`, and also requesting the tournamentDTO returns the correct `metaData`.
> 
> @OllieJennings Can you please provide a tournament code where you're seeing this?

Yep, will send you via discord

**Comment by mariandaniel24** [comment]
I can't seem to be able to whitelist puuid's when creating codes in `tournament-v5`, I'm passing an array to allowedParticipants, however other players (non-whitelisted) can still join the lobby. Is this a bug or am I missing something?

Here's a sample payload I tried:

```json
{
    "enoughPlayers": true,
    "mapType": "SUMMONERS_RIFT",
    "pickType": "TOURNAMENT_DRAFT",
    "spectatorType": "LOBBYONLY",
    "teamSize": 1,
    "allowedParticipants": [...some-puuids]
}
```

**Comment by R0kiee** [comment]
You need to add metaData fiele> I can't seem to be able to whitelist puuid's when creating codes in `tournament-v5`, I'm passing an array to `allowedParticipants`, however other players (non-whitelisted) can still join the lobby. Is this a bug or am I missing something?
> 
> 
> 
> Here's a sample payload I tried:
> 
> 
> 
> ```json
> 
> {
> 
>     "enoughPlayers": true,
> 
>     "mapType": "SUMMONERS_RIFT",
> 
>     "pickType": "TOURNAMENT_DRAFT",
> 
>     "spectatorType": "LOBBYONLY",
> 
>     "teamSize": 1,
> 
>     "allowedParticipants": [...some-puuids]
> 
> }
> 
> ```

**Comment by tisbells** [comment]
> Hey @tisbells, any update on the SEA servers that were introduced in January [last year](https://www.riotgames.com/en/news/new-riot-games-sea-servers-for-lol-tft-go-live-in-january)? We're still getting "invalid region" when setting up tournaments for the Philippines, Singapore, Thailand, Taiwan, and Vietnam. Let me know if you have an ETA for this!

@alpenhannes Hey there, some work for adding these regions began in early December but wasn't able to be completed before the break. I'll see where the team left off and if there's an updated ETA.

**Comment by LeonBirkel** [comment]
> Reposting this here from #833 - We now have a new endpoint for getting the PUUIDs for Tournaments! https://developer.riotgames.com/apis#tournament-v5/GET_getGames
> 
> This will provide the same payload of tournament callbacks + participant PUUIDs. You can also use this endpoint to check if the game was recorded and help debug callbacks. If the endpoint returns the game, it means a callback was attempted. Please let us know if any of you run into problems with this.

@tisbells I do not receive any callbacks since october, I looked to this thread to get everything working again. 
If I call this endpoint I sadly always get the repsonse: "You have been logged out!"
Since yesterday, players also report that they cannot use the tournament codes anymore to join a match.

**Comment by tisbells** [comment]
@LeonBirkel please submit your callback URL, any recent tournament codes and details here - https://support-developer.riotgames.com/hc/en-us

We've received several reports about tournament codes bugging the client and this is currently under investigation. Will post here once we have answers.

**Comment by mrmcnguyen** [comment]
@tisbells I was wondering why "CANNOT EXECUTE. THIS API ENDPOINT IS NOT AVAILABLE IN YOUR POLICY" is showing up when trying to use Tournament-V5?

**Comment by bangingheads** [comment]
> I was wondering why "CANNOT EXECUTE. THIS API ENDPOINT IS NOT AVAILABLE IN YOUR POLICY" is showing up when trying to use Tournament-V5? Any help would be much appreciated :)

Tournament API requires a production application with tournament access specifically requested

**Comment by mrmcnguyen** [comment]
> > I was wondering why "CANNOT EXECUTE. THIS API ENDPOINT IS NOT AVAILABLE IN YOUR POLICY" is showing up when trying to use Tournament-V5? Any help would be much appreciated :)
> 
> Tournament API requires a production application with tournament access specifically requested

I see. In the mean time is Tournament Stubs an alternative? It seems to return a 500 - Internal Server Error

**Comment by jonasdaniels** [comment]
> > Hey @tisbells, any update on the SEA servers that were introduced in January [last year](https://www.riotgames.com/en/news/new-riot-games-sea-servers-for-lol-tft-go-live-in-january)? We're still getting "invalid region" when setting up tournaments for the Philippines, Singapore, Thailand, Taiwan, and Vietnam. Let me know if you have an ETA for this!
> 
> @alpenhannes Hey there, some work for adding these regions began in early December but wasn't able to be completed before the break. I'll see where the team left off and if there's an updated ETA.

Hey @tisbells, any updated ETA on this?

**Comment by tisbells** [comment]
@jonasdaniels we'll be able to circle back to this once we've confirmed the issues with tournament codes (see [issue 888](https://github.com/RiotGames/developer-relations/issues/888)) are resolved. I did remind the team of this request today so once we're squared away with the current issues they'll look at the timeline for adding these regions.  Sorry I can't provide a more concrete ETA just yet

**Comment by Pikeman** [comment]
@tisbells what's the state of `TournamentCodeParametersV5.allowedParticipants`? We observe that the list is _not in effect_ when players join with the tournament code, so the wrong game accounts can end up joining and playing the game.

**Comment by OllieJennings** [comment]
@tisbells We just had a tournament where no tournament callback was received, is there any issues on them being sent?

**Comment by iDanielWongi** [comment]
> @tisbells We just had a tournament where no tournament callback was received, is there any issues on them being sent?

same.

**Comment by Dallin-Cawley** [comment]
Me as well. It doesn't look like there are even attempts. Here's my tournamentCode if it helps: NA04dcb-5f8bb55a-accc-488f-a3df-ceb07dc1d9b9

**Comment by tisbells** [comment]
Hey folks, sorry for the delayed response - The team who owns this service deployed a fix a couple of weeks ago, but I'll let them know there are still issue reports. I'll post back here once I have an update.

**Comment by tisbells** [comment]
The team isn't immediately seeing anything wrong - Could those experiencing this issue please send the following to our [support site](https://support-developer.riotgames.com/):

- Recent tournament codes
- The provider URL
- Confirmation of whether the game looks as expected using [Get Game](https://developer.riotgames.com/apis#tournament-v5/GET_getGames)

**Comment by OllieJennings** [comment]
@tisbells we have had around 5 recent tournaments where not a single code response is being sent from Riot, will send all of the details to Riot as stated above.

**Comment by OllieJennings** [comment]
@tisbells Okay so after raising a ticket the issue is fixed. Basically l had to create a new provider and use HTTPS this time. Now my CA has a cert newer than 2012, but it worked without any issue. So for others, you can use newer CA certs FYI.

**Comment by tisbells** [comment]
Yep, confirming we're being told that if you're having errors with HTTP, you should swap to HTTPS. We know our docs contradict this, but we're working with the folks on the tournament team right now to make sure we have all the right info and will get our docs updated as soon as we can!

**Comment by OllieJennings** [comment]
@tisbells I've raised a support ticket, but just mentioning here.

We are seeing quite a high number of "tournament code is invalid" errors, where simply recreating another code with the exact same parameters work.

We're seeing around 10%-15% of all codes created resulting in this.

**Comment by tisbells** [comment]
@OllieJennings Thanks, we're taking a look! Will post back here once we have additional info.

**Comment by Pikeman** [comment]
> Hi - @Pikeman The team is aware of this issue and it will get fixed, but they are unfortunately unable to provide a timeline just yet. I'll follow up here once they have an update.

Hi @tisbells, just checking in to see if there are any news on `TournamentCodeParametersV5.allowedParticipants`?

**Comment by carlosviol** [comment]
> > Hi - [@Pikeman](https://github.com/Pikeman) The team is aware of this issue and it will get fixed, but they are unfortunately unable to provide a timeline just yet. I'll follow up here once they have an update.
> 
> Hi [@tisbells](https://github.com/tisbells), just checking in to see if there are any news on `TournamentCodeParametersV5.allowedParticipants`?

@tisbells up, same question here.

**Comment by raimannma** [comment]
tisbells no longer works for riot

**Comment by carlosviol** [comment]
@raimannma do you know who can help us about it?

**Comment by raimannma** [comment]
No, I don't know who to contact for LoL API now. 😔

**Comment by OllieJennings** [comment]
@raimannma  I feel you, these days l have resorted to support tickets, but those take like 2 weeks to get a single response. I am still having issues with callbacks not coming in...

**Comment by DaBenjamins** [comment]
Lobbies created with Tournament-V5 can not be started. button is greyed out.

**Comment by Vermillion94** [comment]
Ran into a weird bug today:

I'm getting callbacks to my url for matches played w/ tournament codes, including a match id, but then those matches don't exist in match-v5 (get 404 for NA1_555024955 and NA1_5550195723). 

The tournament code --> match pipeline was working fine yesterday, but is consistently failing tonight; was able to reproduce the behavior w/ a me vs. bots game generated within the same tournament ID.

**Comment by DaBenjamins** [comment]
> Ran into a weird bug today:
> 
> I'm getting callbacks to my url for matches played w/ tournament codes, including a match id, but then those matches don't exist in match-v5 (get 404 for NA1_555024955 and NA1_5550195723).
> 
> The tournament code --> match pipeline was working fine yesterday, but is consistently failing tonight; was able to reproduce the behavior w/ a me vs. bots game generated within the same tournament ID.

I see something similar for games played on tournament codes on the 28th.

**Comment by SoNiice** [comment]
> Ran into a weird bug today:
> 
> I'm getting callbacks to my url for matches played w/ tournament codes, including a match id, but then those matches don't exist in match-v5 (get 404 for NA1_555024955 and NA1_5550195723).
> 
> The tournament code --> match pipeline was working fine yesterday, but is consistently failing tonight; was able to reproduce the behavior w/ a me vs. bots game generated within the same tournament ID.

Can confirm, worked flawlessly til 04/29/26 12PM CEST, didn't work no more at 04/29/26 3PM somehow. Callbacks are triggered, I see the game ending data in my storage, requesting game data through match api results in a 404 error tho.

**Comment by ruuffian** [comment]
Can confirm the above issues with MATCH-V5 on games generated with tournament codes. Some additional colour- the games don't even appear in the client match history.

**Comment by werlend** [comment]
Can confirm. Happened to all our matches starting april 29th

**Comment by SundayKoi** [comment]
is there any known plan to fix this or have they said anything about it?

**Comment by IbaiMontero** [comment]
Still broken i cant get data from my tournament codes games....
